lib/galaxy/datatypes/tabular.py

```diff
@@ -691,6 +695,23 @@ def set_meta( self, dataset, **kwd ):
             # Found header line, get sample names.
             dataset.metadata.sample_names = line.split()[ 9: ]
 
+    @staticmethod
+    def merge(split_files, output_file):
+        tmp_dir = tempfile.mkdtemp()
+        stderr_name = tempfile.NamedTemporaryFile(dir=tmp_dir, prefix="bam_merge_stderr").name
+        command = ["bcftools", "concat"] + split_files + ["-o", output_file]
+        log.info("Merging vcf files with command [%s]" % " ".join(command))
+        proc = subprocess.Popen( args=command, stderr=open( stderr_name, 'wb' ) )
+        exit_code = proc.wait()
+        # Did merge succeed?
+        stderr = open(stderr_name).read().strip()
+        if stderr:
+            if exit_code != 0:
+                shutil.rmtree(tmp_dir)  # clean up
```

Comment:
Both if's can be combined  and shutil.rmtree() can be used instead of os.unlink() at the bottom.